src/include/daos_srv/bio.h

```diff
@@ -310,6 +310,47 @@ bio_sgl_holes(struct bio_sglist *bsgl)
 	return result;
 }
 
+enum bio_dev_state {
+	BIO_DEV_NORMAL	= 0, /* fully functional and in-use */
+	BIO_DEV_FAULTY,      /* evicted device */
+	BIO_DEV_OUT,         /* unplugged device */
+	BIO_DEV_NEW,         /* new device not currently in-use */
+};
```

Comment:
These are all "displayed" state, I don't think it's required in "bio.h". I don't think it's quite necessary to define these displayed state, if it's useful for control plane, please move the definition to control plane.

Reply:
these constants could be defined in control/server/storage/types.go {look at ScmState* for an example) then they will be able to be imported from both server and dmg so you can set state based on dRPC response in the server (bdi_flags uint8) serialize state in grpc response and unpack in dmg, adding a stringer interface implementation to the state type in types.go will then enable you to print a string representation of the state received in dmg